spyql/log.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import asciichartpy as colors
+import sys
+import json
+
+error_on_warning = False
+
+def user_error(message, exception, code = None, vars = None):
+    """
+    Reports an error, throwing the original exception
+    Prints a custom message.
+    Prints the data that originated the exception (if available).
+    """
+    sys.tracebacklimit = 0
+    logging.error(f"{colors.lightred}ERROR{colors.reset}\t{message}{': ' + colors.lightyellow + str(code) + colors.reset if code else ''}")
```

Comment:
Does this conditional apply to the entire `': ' + colors.lightyellow + str(code) + colors.reset`?

Reply:
Yes. Basically returns an empty string if code is not present, otherwise adds ':' followed by ' 'yellow' code.